### PR TITLE
[Doctrine] Fix repetitive assigns definition on InitializeDefaultEntityCollectionRector

### PR DESCRIPTION
**Pull request by samsonasik**
Fix repetitive assign definition on auto import enabled Fix https://github.com/rectorphp/rector/issues/6697

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: src/Rector/Class_/InitializeDefaultEntityCollectionRector.php [hunk 1]
```diff
@@ -22,6 +22,11 @@
  */
 final class InitializeDefaultEntityCollectionRector extends AbstractRector
 {
+    /**
+     * @var array<string, bool>
+     */
+    private $filesApplied = [];
```
Comment (TomasVotruba) [inline src/Rector/Class_/InitializeDefaultEntityCollectionRector.php:28]: Should by `array` typee

Comment (samsonasik) [reply src/Rector/Class_/InitializeDefaultEntityCollectionRector.php]: Updated with array type

**Comment by TomasVotruba** [comment]
What about cases, when the init asking is already there?
I think the check should look for existing assign, not for modified file.

**Comment by samsonasik** [comment]
It's assign exists already there, the check assign exists is called repetitively, I am not sure why, possibly need to be checked on rector-src single rule should apply to a file once?

**Comment by samsonasik** [comment]
@TomasVotruba it seems can be handled in `AbstractRector`, something like this:

```php
        if (isset($this->appliedRulesOnFile[$this::class][$this->file->getSmartFileInfo()->getRealPath()])) {
            return true;
        }

        $this->appliedRulesOnFile[$this::class][$this->file->getSmartFileInfo()->getRealPath()] = true;
        return false;
```

I will create alternative PR to rector-src for it.

**Comment by TomasVotruba** [comment]
That would break applying Rector in multiple runs.

E.g. the rule that remove `true || true`, should only work with one node. Change on first run:

```diff
-if (true || true || true || true || true || true) {
+if (true || true || true || true || true) {
```

Change on second run:

```diff
-if (true || true || true || true) {
+if (true || true || true) {
```

etc.

**Comment by TomasVotruba** [comment]
Here the check should be improved only in this specific rule, looking for existing stmts probably. Or if these are not availables, use an `$node->setAttribute()` marker as other rules use.

**Comment by samsonasik** [comment]
Ok, I will try.

**Comment by samsonasik** [comment]
Using `spl_object_hash` seems did a trick https://github.com/rectorphp/rector-doctrine/pull/28/commits/0eef46566c984c07007070a5aae228a3f175280d

**Comment by samsonasik** [comment]
or not, I will re-check.

**Comment by samsonasik** [comment]
Fixed 🎉 , I updated with check `$node->getAttribute(AttributeKey::KIND)` check against value "initialized"

**Comment by TomasVotruba** [comment]
Looks good, thanks :+1:

**Comment by samsonasik** [comment]
@TomasVotruba can we have new rector-doctrine release for this patch so it will be included in next rector release?

**Comment by TomasVotruba** [comment]
Sure. Can you tag it?

**Comment by samsonasik** [comment]
Yes, I tagged https://github.com/rectorphp/rector-doctrine/releases/tag/0.11.18

**Comment by TomasVotruba** [comment]
You can do same for Symfony

**Comment by samsonasik** [comment]
Ok, tagged for symfony https://github.com/rectorphp/rector-symfony/releases/tag/0.11.19

**Pull request merged**